src/main/java/mate/academy/dao/MovieDaoImpl.java

```diff
@@ -0,0 +1,54 @@
+package mate.academy.dao;
+
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.models.Movie;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.HibernateException;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+import java.util.Optional;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+    private static final SessionFactory sessionFactory = HibernateUtil.getSessionFactory();
+
+    @Override
+    public Movie add(Movie movie) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
```

Comment:
optional, u can use static import for method getSessionFactory() and use it like this
```suggestion
    @Override
    public Movie add(Movie movie) {
        Session session = null;
        Transaction transaction = null;
        try {
            session = getSessionFactory().openSession();
```